jitbuilder/release/README.md

```diff
@@ -20,25 +20,118 @@ OpenJDK Assembly Exception [2].
+The JitBuilder API also includes a few Vector services for generating
+vector instructions and even has a `Transaction` service that makes it
+simple to build code that takes advantage of hardware transactional
+memory.
 
-Just type 'make' and you'll build a number of examples that
-demonstrate how you can start building native code dynamically!
+And if your goal is to create a JIT compiler for a virtual machine for
+some language, the JitBuilder API has services to help you efficiently
+model pushes and pops on operand stacks or to access registers in an
+operand array that frequently allocate your values to hardware registers
+rather than memory locations. There have even been experiments that look
+at automatically generating both an interpreter and a JIT compiler for
+simple virtual machines using just the definitions of the virtual
+machine (VM) state and descriptions of the operations to be performed
+by the VM's bytecodes.
 
-More details to come!
+JitBuilder can generate efficient native code on X86, POWER, and Z platforms.
+Multiple experiments have found the generated JitBuilder code performance
+to be comparable or, in some cases, even better than what LLVM can do!
+Because it's based on the OMR compiler technology at the heart of the
+[Eclipse OpenJ9](https://www.eclipse.org/openj9), it uses memory
+efficiently and compiles quickly.
 
-This library has been used as the basis of a JIT compiler for
-the SOM++ Simple Object Machine runtime.
+# Steps to build
+
+Here are the basic steps required to build the JitBuilder library and
+run the 28 code samples from the C++ language.
+
+First, clone the Eclipse OMR repository:
+
+```
+$ git clone https://github.com/eclipse/omr
+```
+
+Change directory into the OMR project, create a `build` directory, and
+generate the makefiles using `cmake` :
+
+```
+$ cd omr
+$ mkdir build
+$ cd build
+$ cmake .. -DOMR_COMPILER=1 -DOMR_JITBUILDER=1 -DOMR_TEST_COMPILER=1
+```
+
+Now, you're ready to build all the OMR componentry and create the
+JitBuilder static library. This step may take a few minutes! It may
+go faster if you use `make -jN` with `N` set to the number of cores
+on your machine but you can always just type `make` to build the
+library:
+
+```
+$ make
+```
+
+Now that the JitBuilder library has been built, you can change directory
+into the `jitbuilder/release` directory which will eventually have different
+subdirectories for using the JitBuilder API from different languages. We're
+going to use C++, so we'll go into the `cpp` subdirectory:
+
+```
+$ cd ../jitbuilder/release/cpp
+```
+
+You'll need to create a link to the `libjitbuilder.a` library that you
+built. This step should go away some day, but for now you'll need to do
+it manually:
+
+```
+$ ln -s ../../../build/jitbuilder/libjitbuilder.a .
+```
+
+Finally, you can build and run the code samples by running:
+
+```
+$ make test
+```
+
+If you're running on an x86 platform, then you should be able to run all
+28 code samples with:
```

Comment:
This number looks like something that will quickly go out of sync. Up to you whether we want to omit it.

Reply:
reasonable point, though it hasn't been growing much lately :) . Updated the wording.